### ISSUE TITLE
Inconsistent definition of copy! for PETScVector ?

### ISSUE THREAD

**Issue by fverdugo** [gridap/GridapPETSc.jl#50]
I have the feeling that the definition of these two functions:

https://github.com/gridap/GridapPETSc.jl/blob/3ba0de4ecd9c04ed1203f972b2c0409981aa7b5e/src/PETScArrays.jl#L121
https://github.com/gridap/GridapPETSc.jl/blob/3ba0de4ecd9c04ed1203f972b2c0409981aa7b5e/src/PETScArrays.jl#L136

is inconsistent with the Julia definition of `copy!`:

```julia
help?> copy!
search: copy! copyto! circcopy! unsafe_copyto! copy copysign deepcopy

  copy!(dst, src) -> dst

  In-place copy of src into dst, discarding any pre-existing elements in dst.
  If dst and src are of the same type, dst == src should hold after the call.
  If dst and src are multidimensional arrays, they must have equal axes. See
  also copyto!.
```

In particular, with the sentence "if dst and src are multidimensional arrays, they must have equal axes. " As far as I can tell this means that `length(dst) == length(src)` should be true. And I think that this is not true in the functions linked above. In other words, both vectors need to be understood either as global vectors, or as local ones.

@amartinhuertas Do you remember why you needed to change these functions?

**Comment by amartinhuertas** [comment]
> @amartinhuertas Do you remember why you needed to change these functions?

I did not change them, I added them. They are new. I added them along with their two counterparts in `PartitionedArrays.jl` for the `PVector,PETScVector` combinations, and vice-versa.

These are needed in order to copy from PETSc vectors to the vectors that were used to assemble the system, either `PVector` or plain Julia `Vector`s.

Does all this answer your question?
